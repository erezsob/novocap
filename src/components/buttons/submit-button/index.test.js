import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SubmitButton from './';

describe('<SubmitButton />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SubmitButton />);
  });

  it('renders', () => {
    expect(wrapper).toHaveLength(1);
  });

  describe('renders properly', () => {
    const tree = renderer.create(<SubmitButton />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('has the correct type attribute', () => {
    expect(wrapper.dive().find('[type="submit"]')).toHaveLength(1);
  });
});
