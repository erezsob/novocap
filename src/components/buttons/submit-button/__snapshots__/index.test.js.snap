// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="MuiButtonBase-root-17 MuiButton-root-2 SubmitButton-submit-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="submit"
>
  <span
    className="MuiButton-label-3"
  >
    Save
  </span>
  <span
    className="MuiTouchRipple-root-20"
  />
</button>
`;
